novawallet/Modules/DApp/DAppBrowser/DAppBrowserPresenter.swift

```diff
@@ -51,6 +56,20 @@ final class DAppBrowserPresenter {
 
         view?.didSet(canShowSettings: canShowSettings)
     }
+
+    private func saveCurrentWebViewState() {
+        if
+            let currentTabId,
+            let currentWebView = webViewPool.getWebView(for: currentTabId),
```

Comment:
I would probably pass pool directly to the controller via init. And controller can call presenter to save opaque state given in the parameters